samples/ImageScanning/cpp/ImageScanningSample_Desktop/main.cpp

```diff
@@ -0,0 +1,288 @@
+
+using namespace Microsoft::AI::Skills::SkillInterfacePreview;
+using namespace Microsoft::AI::Skills::Vision::ImageScanningPreview;
+
+// enum to string lookup table for SkillExecutionDeviceKind
+static const std::map<SkillExecutionDeviceKind, std::string> SkillExecutionDeviceKindLookup = {
+    { SkillExecutionDeviceKind::Undefined, "Undefined" },
+    { SkillExecutionDeviceKind::Cpu, "Cpu" },
+    { SkillExecutionDeviceKind::Gpu, "Gpu" },
+    { SkillExecutionDeviceKind::Vpu, "Vpu" },
+    { SkillExecutionDeviceKind::Fpga, "Fpga" },
+    { SkillExecutionDeviceKind::Cloud, "Cloud" }
+};
+
+// enum to string lookup table for ImageRectifierInterpolationKind
+static const std::map<ImageRectifierInterpolationKind, std::string> ImageRectifierInterpolationKindLookup = {
+    { ImageRectifierInterpolationKind::Bilinear, "Bilinear" },
+    { ImageRectifierInterpolationKind::Bicubic, "Bicubic" }
+};
+
+// enum to string lookup table for ImageCleaningKind
+static const std::map<ImageCleaningKind, std::string> ImageCleaningKindLookup = {
+    { ImageCleaningKind::WhiteboardOrDocument, "WhiteboardOrDocument" },
+    { ImageCleaningKind::Whiteboard, "Whiteboard" },
+    { ImageCleaningKind::Document, "Document" },
+    { ImageCleaningKind::Picture, "Picture" },
+};
+
+//
+// Load a VideoFrame from a specified image file path
+//
+VideoFrame LoadVideoFrameFromImageFile(hstring imageFilePath)
+{
+    VideoFrame resultFrame = nullptr;
+    SoftwareBitmap softwareBitmap = nullptr;
+    try
+    {
+        StorageFile file = StorageFile::GetFileFromPathAsync(imageFilePath).get();
+        auto fileExtension = file.FileType();
+        if (fileExtension != L".jpg" && fileExtension != L".png")
+        {
+            throw hresult_invalid_argument(L"This app parses only .png and .jpg image files");
+        }
+
+        IRandomAccessStream stream = file.OpenAsync(FileAccessMode::Read).get();
+
+        // Create the decoder from the stream 
+        BitmapDecoder decoder = BitmapDecoder::CreateAsync(stream).get();
+
+        // Get the SoftwareBitmap representation of the file in BGRA8 format
+        softwareBitmap = decoder.GetSoftwareBitmapAsync().get();
+
+        // Convert to friendly format for UI display purpose
+        softwareBitmap = SoftwareBitmap::Convert(softwareBitmap, BitmapPixelFormat::Bgra8, BitmapAlphaMode::Premultiplied);
+
+        // Encapsulate the image in a VideoFrame instance
+        resultFrame = VideoFrame::CreateWithSoftwareBitmap(softwareBitmap);
+
+        stream.Close();
+    }
+    catch (hresult_error const& ex)
+    {
+        std::wcerr << "Could not load VideoFrame from file: " << imageFilePath.c_str() << std::endl;
+        throw ex;
+    }
+    return resultFrame;
+}
+
+//
+// Save a modified VideoFrame using an existing image file path with an appended suffix
+//
+hstring SaveModifiedVideoFrameToFile(hstring imageFilePath, VideoFrame frame)
+{
+    hstring fileName = imageFilePath;
```

Comment:
nit: this is unnecessary.

You can either use `imageFilePath` directly (as you're passing by copy) or just return `modifiedFile.Path()` from within the `try` block